contracts/modules/Experimental/Checkpoints/PLCRVotingCheckpoint.sol

```diff
@@ -0,0 +1,313 @@
+pragma solidity ^0.5.0;
+
+import "../../Module.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract PLCRVotingCheckpoint is Module {
+
+    using SafeMath for uint256;
+
+    enum Stage { PREP, COMMIT, REVEAL, RESOLVED }
+
+    struct Ballot {
+        uint256 quorum;       // Should be a multiple of 10 ** 16
+        uint256 totalSupply;
+        uint256 checkpointId;
+        uint64 commitDuration;
+        uint64 revealDuration;
+        uint64 startTime;
+        uint24 totalProposals;
+        uint32 totalVotes;
+        uint8 isActive;
+        mapping(uint256 => uint256) weightedVote;
+        mapping(address => Vote) proposalToVote;
+    }
+
+    struct Vote {
+        uint256 voteOption;
+        bytes32 secretVote;
+        bool isRevealed;
```

Comment:
We don't need `isRevealed`. Once the voter reveals their vote, we should delete `secretVote`. This will make both commit and reveal transactions cheaper.